Hi!

Currently, when both S4 and S4bios are available, swsusp always
chooses S4bios and makes S4 unavailable. Bad idea as S4bios needs
completely different setup.... Please apply,
								Pavel

--- clean-mm/drivers/acpi/sleep/main.c	2004-08-17 12:21:43.000000000 +0200
+++ linux-mm/drivers/acpi/sleep/main.c	2004-08-17 12:22:45.000000000 +0200
@@ -217,7 +217,8 @@
 				sleep_states[i] = 1;
 				printk(" S4bios");
 				acpi_pm_ops.pm_disk_mode = PM_DISK_FIRMWARE;
-			} else if (sleep_states[i])
+			} 
+			if (sleep_states[i])
 				acpi_pm_ops.pm_disk_mode = PM_DISK_PLATFORM;
 		}
 	}

